### PR TITLE
Merge DM dev docs to TiDB dev docs

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

To help users access TiDB Data Migration (DM) documentation more quickly and conveniently, this PR merges TiDB DM dev docs into TiDB dev docs. 

The major changes are as follows:
1. Copy the DM dev documentation in https://github.com/pingcap/docs-dm/pull/869/files to the TiDB dev documentation, and add the DM dev TOC to TiDB dev TOC.
2. In DM dev documentation, update the external links of TiDB documentation to internal links.
3. Update the external links of DM documentation to internal links. 
4. Add dm-releases-notes.md. 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -198,6 +198,91 @@
     - [FAQ](/tidb-lightning/tidb-lightning-faq.md)
     - [术语表](/tidb-lightning/tidb-lightning-glossary.md)
   - [TiDB Data Migration](/dm/dm-overview.md)
```
Comment (TomShawn) [inline TOC.md:200]:
```suggestion
  - TiDB Data Migration
```

File: dm/dm-enable-tls.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    from:
+        security:
+            ssl-ca: "/path/to/mysql-ca.pem"
+            ssl-cert: "/path/to/mysql-client-cert.pem"
+            ssl-key: "/path/to/mysql-client-key.pem"
+    ```
+
+### 为下游 TiDB 连接开启加密传输
+
+1. 配置下游 TiDB 启用加密连接支持，具体可参考 [配置 TiDB 启用加密连接支持](https://docs.pingcap.com/zh/tidb/stable/enable-tls-between-clients-and-servers#配置-tidb-启用加密连接支持)
```
Comment (TomShawn) [inline dm/dm-enable-tls.md:113]:
```suggestion
1. 配置下游 TiDB 启用加密连接支持，具体可参考 [配置 TiDB 启用加密连接支持](/enable-tls-between-clients-and-servers#配置-tidb-服务端启用安全连接)
```

File: dm/dm-faq.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+5. 通过 `query-status` 观察迁移任务状态，当 `syncerBinlog` 超过 `checkpoint-T` 与 `checkpoint-S` 中的较大值后（在本例中，为 `(mysql-bin.000100, 1234)`），即可还原 `safe-mode` 为原始值并重启迁移任务。
+
+## 全量导入过程中遇到报错 `packet for query is too large. Try adjusting the 'max_allowed_packet' variable`
+
+尝试将
+
+- TiDB Server 的全局变量 `max_allowed_packet`
+- 任务配置文件中的配置项 `target-database.max-allowed-packet`（详情参见 [DM 任务完整配置文件介绍](/dm/task-configuration-file-full.md)）
+
+设置为比默认 67108864 (64M) 更大的值。详见 [Loader 解决方案](https://docs.pingcap.com/zh/tidb/stable/loader-overview#解决方案)。
```
Comment (TomShawn) [inline dm/dm-faq.md:119]: Loader 在 v5.0 中就废弃了

File: dm/dm-faq.md [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+可以检查 DM-worker 日志，在其中搜索包含 `change count` 的行，该行的 `new count` 非零时会启用 safe mode。检查 safe mode 启用时间以及启用前是否有报错，以定位启用原因。
+
+## 使用 DM v2.0 同步数据时重启 DM 进程，出现全量数据导入失败错误
+
+在 DM v2.0.1 及更早版本中，如果全量导入操作未完成时发生重启，重启后的上游数据源与 DM worker 的绑定关系可能会发生变化。例如，可能出现 dump 单元的中间数据在 DM worker A 机器上，但却由 DM worker B 进行 load 单元的情况，进而导致操作失败。
+
+该情况有两种解决方案：
+
+- 如果数据量较小（TB 级以下）或任务有合库合表：清空下游数据库的已导入数据，同时清空导出数据目录，使用 dmctl 删除并 `start-task --remove-meta` 重建任务。后续尽量保证全量导出导入阶段 DM 没有冗余 worker 以及避免在该时段内重启或升级 DM 集群。
+- 如果数据量较大（数 TB 或更多）：清空下游数据库的已导入数据，将 lightning 部署到数据所在的 DM worker 节点，使用 [lightning local backend 模式](https://docs.pingcap.com/zh/tidb/dev/deploy-tidb-lightning) 导入 DM dump 单元导出的数据。全量导入完成后，修改任务的 `task-mode` 为 `incremental`，修改 `mysql-instance.meta.pos` 为 dump 单元导出数据 `metadata` 中记录的位置，启动一个增量任务。
```
Comment (TomShawn) [inline dm/dm-faq.md:196]: Global comment: Please check all links that start with `https://docs.pingcap.com/zh/tidb/` and `https://pingcap.com/docs-cn/`.
```suggestion
- 如果数据量较大（数 TB 或更多）：清空下游数据库的已导入数据，将 lightning 部署到数据所在的 DM worker 节点，使用 [lightning local backend 模式](https://docs.pingcap.com/zh/tidb/dev/deploy-tidb-lightning) 导入 DM dump 单元导出的数据。全量导入完成后，修改任务的 `task-mode` 为 `incremental`，修改 `mysql-instance.meta.pos` 为 dump 单元导出数据 `metadata` 中记录的位置，启动一个增量任务。
```

File: dm/dm-open-api.md [hunk 5]
```diff
@@ -0,0 +1,1084 @@
...
+aliases: ['/zh/tidb-data-migration/dev/open-api/']
+---
+
+# 使用 OpenAPI 运维集群
+
+> **警告：**
+>
+> 当前该功能为实验特性，默认关闭，不建议在生产环境中使用。
+
+DM 提供 OpenAPI 功能，你可以通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](/dm/./dmctl-introduction.md)类似。如需开启该功能，请在 DM-master 的配置文件中增加如下配置项：
```
Comment (TomShawn) [inline dm/dm-open-api.md:13]:
```suggestion
DM 提供 OpenAPI 功能，你可以通过 OpenAPI 对 DM 集群进行查询和运维操作。OpenAPI 的总体功能和 [dmctl 工具](/dm/dmctl-introduction.md)类似。如需开启该功能，请在 DM-master 的配置文件中增加如下配置项：
```

File: dm/dm-precheck.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    - REPLICATION CLIENT
+    - RELOAD
+    - SELECT
+
++ 上游 MySQL 表结构的兼容性
+
+    TiDB 和 MySQL 的兼容性存在以下一些区别：
+
+    - TiDB 不支持外键
+    - 字符集的兼容性不同，详见 [TiDB 支持的字符集](https://pingcap.com/docs-cn/dev/reference/sql/character-set/)
```
Comment (TomShawn) [inline dm/dm-precheck.md:52]: ditto

File: dm/dm-faq.md (hunk 3 above)
Comment (qiancai) [reply dm/dm-faq.md:119]: @sunzhaoyang 请帮忙看下是否有 Loader 的替代方案呢？谢谢

File: dm/benchmark-v1.0-ga.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
+---
+title: DM 1.0-GA 性能测试报告
```
Comment (lilin90) [inline dm/benchmark-v1.0-ga.md:2]: 迁过来的新版本的文档里还需要保留这个文档吗？看到 TOC 里也没加相关链接。

File: dm/benchmark-v2.0-ga.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
+---
+title: DM 2.0-GA 性能测试报告
```
Comment (lilin90) [inline dm/benchmark-v2.0-ga.md:2]: 迁过来的新版本的文档里还需要保留这个文档吗？看到 TOC 里也没加相关链接。

File: dm/benchmark-v1.0-ga.md (hunk 7 above)
Comment (qiancai) [reply dm/benchmark-v1.0-ga.md:2]: 我也觉得不用，那我删了这个文件吧。

File: dm/benchmark-v2.0-ga.md (hunk 8 above)
Comment (qiancai) [reply dm/benchmark-v2.0-ga.md:2]: 我也觉得不用，那我删了这个文件吧。

File: dm/dm-faq.md (hunk 3 above)
Comment (glorv) [reply dm/dm-faq.md:119]: 这个 bug 在 dm v2.0 GA 版本就已经修复了。我觉得可以采用如下两个方案：
1. 去掉这个链接，如果还有用老版本的用户遇到这个问题，直接告诉他需要怎么workaround 就够了，不用在强调 loader.
2. 将链接强制跳转到 3.0/4.0 版本的 loader 配置页面，即  https://docs.pingcap.com/zh/tidb/v4.0/loader-overview/

**Review by TomShawn — APPROVED** [review]
LGTM

File: dm/dm-faq.md (hunk 3 above)
Comment (sunzhaoyang) [reply dm/dm-faq.md:119]: 赞同去掉链接。根据前面的信息设置更大的值就可以解决了。

**Review by sunzhaoyang — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
LGTM

File: dm/dm-faq.md (hunk 3 above)
Comment (qiancai) [reply dm/dm-faq.md]: Done.

**Pull request merged**